pkg/controller/versions/strategy_v2_3.go

```diff
@@ -224,6 +273,71 @@ func (v *versionStrategyV2_3) ValidateUpgrade(ctx context.Context, cl client.Cli
 	return nil
 }
 
+func (v *versionStrategyV2_3) ValidateUpdate(ctx context.Context, cl client.Client, oldSMCP, newSMCP metav1.Object) error {
+	oldClusterScoped, err := isClusterScoped(oldSMCP)
+	if err != nil {
+		return err
+	}
+	newClusterScoped, err := isClusterScoped(newSMCP)
+	if err != nil {
+		return err
+	}
+	if oldClusterScoped != newClusterScoped {
+		return fmt.Errorf("field spec.techPreview.%s is immutable", v2.ControlPlaneModeKey)
```

Comment:
What does this error mean? Can it be reworded so that it's easier to understand?

Reply:
That's a standard k8s error. See https://www.datree.io/resources/kubernetes-error-codes-field-is-immutable